src/components/Carousel.tsx

```diff
@@ -1,18 +1,107 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Carousel.scss';
+import cn from 'classnames';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+interface Props {
+  images: string[],
+  step: number,
+  frameSize: number,
+  itemWidth: number,
+  animationDuration: number,
+  infinite: boolean,
+}
+
+const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const gap = 20;
+  const totalGapValue = gap * (frameSize - 1);
+  const visibleImagesWidth = itemWidth * frameSize + totalGapValue;
+
+  const [translateValue, setTranslateValue] = useState(0);
+
+  const newTranslateValue = step * (itemWidth + gap);
+
+  const allImagesWidth = (10 * itemWidth) + (9 * gap);
+  const extremeTranslatePoint = allImagesWidth - visibleImagesWidth;
+
+  const handleeClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    const currentButton = event.currentTarget;
+    const currentButtonClasslist = currentButton.classList;
+    let value = 0;
+
+    if (currentButtonClasslist.contains('button-prev')) {
+      value = Math.min(translateValue + newTranslateValue, 0);
+    } else if (currentButtonClasslist.contains('button-next')) {
+      value = Math.max(
+        translateValue - newTranslateValue, (-extremeTranslatePoint),
+      );
+    }
+
+    if (infinite && -1 * value >= extremeTranslatePoint) {
+      value = 0;
+    }
+
+    setTranslateValue(value);
+  };
+
+  return (
+    <div className="Carousel">
+      <div className="Carousel__container">
+        <button
+          className={cn('button-prev', {
+            'button-prev__disabled': translateValue === 0,
+          })}
+          type="button"
+          onClick={handleeClick}
+        >
+          <div className="button-prev__text">&lt;</div>
+        </button>
+        <ul className="Carousel__list" style={{ width: `${visibleImagesWidth}px` }}>
+          <div
+            className="Carousel__item-wrapper"
+            style={{
+              transform: `translateX(${translateValue}px)`,
+              transition: `transform ${animationDuration}ms ease`,
+              gap: `${gap}px`,
+            }}
+          >
+            {images.map(image => (
+              <li
+                className="Carousel__item"
+                key={image}
+                style={{ height: `${itemWidth}px` }}
+              >
```

Comment:
```suggestion
          <div
            className="Carousel__item-wrapper"
            style={{
              transform: `translateX(${translateValue}px)`,
              transition: `transform ${animationDuration}ms ease`,
              gap: `${gap}px`,
            }}
          >
            {images.map(image => (
              <li
                className="Carousel__list-item"
                key={image}
                style={{
                  transform: `translateX(${translateValue}px)`,
                  transition: `transform ${animationDuration}ms ease`,
                }}
              >
```
it and width for img should fix at least this two tests
![Screenshot from 2023-08-12 16-45-54](https://github.com/mate-academy/react_carousel/assets/33029864/def4f28a-9957-4241-beed-5ebaab5bc20a)